perun-rpc/src/main/java/cz/metacentrum/perun/rpc/Api.java

```diff
@@ -834,6 +842,71 @@ private Deserializer selectDeserializer(String format, HttpServletRequest req) t
 		}
 	}
 
+	private record OidcSubIssuer(String sub, String issuer) {}
+
+	private static OidcSubIssuer initOidcAuth(HttpServletRequest req, Map<String, String> additionalInformations) {
+		var accessToken = req.getHeader(OIDC_ACCESS_TOKEN);
+		var restTemplate = new RestTemplate();
+		var issuer = req.getHeader(OIDC_CLAIM_ISS);
+		var config = restTemplate.getForObject(issuer + "/.well-known/openid-configuration", JsonNode.class);
+		var headers = new HttpHeaders();
+		headers.setBearerAuth(accessToken);
+		HttpEntity<Void> entity = new HttpEntity<>(headers);
+		var userInfoResponse = restTemplate.exchange(config.path("userinfo_endpoint").textValue(), HttpMethod.GET, entity, JsonNode.class);
+		var userInfo = userInfoResponse.getBody();
+		if(isNotEmpty(userInfo.path("error").asText())) {
+			log.error("Call to user info endpoint failed, the error is: {}", userInfo);
+			throw new InternalErrorException("Call to user info endpoint failed, the error is" + userInfo);
+		}
+		log.debug("user info retreated: {}", userInfo);
+		var name = userInfo.path("name").asText();
+		if(isEmpty(name)) {
+			var firstName = userInfo.path("given_name").asText();
+			var familyName = userInfo.path("family_name").asText();
+			if(isNotEmpty(firstName) && isNotEmpty(familyName)) {
+				name =  firstName + " " + familyName;
+			}
+		}
+		var email = userInfo.path("email").asText();
+		if(isNotEmpty(email)) {
+			additionalInformations.put("mail", email);
+		}
+		if(isNotEmpty(name)) {
+			additionalInformations.put("displayName", name);
+		}
```

Comment:
adding to additional informations display name and mail - discuss if we want more in the additional information